crates/matrix-sdk/src/http_client.rs

```diff
@@ -137,27 +137,22 @@ impl HttpClient {
         config: RequestConfig,
     ) -> Result<http::Request<Bytes>, HttpError> {
         let read_guard;
-        let access_token = if config.force_auth {
-            read_guard = session.read().await;
-            if let Some(session) = read_guard.as_ref() {
-                SendAccessToken::Always(session.access_token.as_str())
-            } else {
-                return Err(HttpError::ForcedAuthenticationWithoutAccessToken);
```

Comment:
This error is now dead code, let's remove it.

Reply:
Sorry, I wasn't clear. I meant that we should remove it completely, from the `HttpError` enum.